### PR TITLE
Revise Troubleshooting.md

### PR DESCRIPTION
**Pull request by ddbeck**
After having a bit of difficulty with the troubleshooting steps in [a previous issue](https://github.com/Homebrew/brew/pull/1437), I've tried to revise the _Troubleshooting_ document to:

* make the doc easier to follow top to bottom (e.g., putting troubleshooting steps in order of specificity, putting issue contents guidelines after all of the troubleshooting steps, and adding some section signposting)
* rephrase things to capture formula-related problems that are not explicitly build failures

I also tweaked a few things here and there to be (hopefully) more readable. Thanks for considering my PR!

----

- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [N/A] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [X] Have you successfully run `brew tests` with your changes locally?

### CODE REVIEW

File: docs/Troubleshooting.md [hunk 1]
```diff
@@ -1,26 +1,34 @@
 # Troubleshooting
-**Run `brew update` (twice) and `brew doctor` *before* creating an issue!**
```
Comment (MikeMcQuaid) [inline docs/Troubleshooting.md]: Would rather leave this at the top as people often don't read more than the first line here, sadly.

**Comment by scpeters** [comment]
I agree, thanks for clarifying the debugging steps!

**Comment by ddbeck** [comment]
OK, I've brought back the attention-grabbing first line and tweaked the introduction to match up with that a little better. Let me know if you want to see any other changes. Thanks!

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request merged**